csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs

```diff
@@ -889,14 +932,196 @@ public static OrtValue CreateMap(OrtValue keys, OrtValue values)
         {
             if (keys is null || values is null)
             {
-                throw new ArgumentNullException($"keys or/and values are null");
+                throw new ArgumentNullException("keys or/and values are null");
             }
 
+            var compositeMembers = new DisposableList<OrtValue>
+            {
+                keys,
+                values
+            };
+
             IntPtr[] handles = { keys.Handle, values.Handle };
             NativeApiStatus.VerifySuccess(
                 NativeMethods.OrtCreateValue(handles, (UIntPtr)handles.Length, (IntPtr)OnnxValueType.ONNX_TYPE_MAP,
                                out IntPtr mapHandle));
-            return new OrtValue(mapHandle, OnnxValueType.ONNX_TYPE_MAP);
+            return new OrtValue(mapHandle, OnnxValueType.ONNX_TYPE_MAP, compositeMembers);
+        }
+
+        /// <summary>
+        /// Creates a map OrtValue with keys and values specified as arrays.
+        /// This helps the user not to create OrtValues for keys and values separately.
+        /// 
+        /// The OrtValues would be created on top of the managed memory using it directly.
+        /// The number of elements in keys and values must be the same and they must be in order.
+        /// 
+        /// The types must be unmanaged.
+        /// </summary>
+        /// <typeparam name="K">keys type</typeparam>
+        /// <typeparam name="V">values type</typeparam>
```

Comment:
Is this the element type for the data that will be created in an OrtValue containing a Tensor?

Reply:
Will clarify the documentation